Add ceed-* devices to test harness

lindsayad opened the pull request:
Refs #30915 and #31126

If/when this is merged I think I'll add `ceed-cpu` and `ceed-cuda` steps to some civet recipe. The latter can certainly go in our CUDA GPU recipe, but I'd have to think about where `ceed-cpu` should go. Currently both will be empty, but `ceed-cpu` will have a couple of associated tests after #31126

lindsayad committed:
Add ceed-* devices to test harness

Refs #39015

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/3075354/), step Docs: sync website on ab0e742 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/31151/site)

This comment will be updated on new commits.

nmnobre commented on python/TestHarness/TestHarness.py:
```diff
@@ -1139,7 +1143,7 @@ def parseCLArgs(self, argv):
         parser.add_argument('--parallel-mesh', action='store_true', dest='parallel_mesh', help='Deprecated, use --distributed-mesh instead')
         parser.add_argument('--distributed-mesh', action='store_true', dest='distributed_mesh', help='Pass "--distributed-mesh" to executable')
-        parser.add_argument('--device', action='store', dest='device', type=str, choices=['cpu', 'cuda', 'hip', 'mps'], default='cpu', help='Run libtorch or MFEM tests with this device')
+        parser.add_argument('--device', action='store', dest='device', type=str, choices=TestHarness.validDevices(), default='cpu', help='Run libtorch or MFEM tests with this device')
```
```suggestion
        parser.add_argument('--device', action='store', dest='device', type=str, choices=TestHarness.validDevices(), default='cpu', help='Run libtorch or MFEM tests with this device; device availability depends on library support and compilation settings')
```

lindsayad committed:
Apply suggestions from Nuno

- Add missing quotation mark
- more verbose argument doc string regarding device availability/support

Co-authored-by: Nuno Nobre <nuno.nobre@stfc.ac.uk>

lindsayad commented:
@loganharbour sha help plz

loganharbour reviewed (approved):
I think that this can go in standalone as is?

loganharbour commented:
HPC failure is unrelated; trust added
